datalad_next/constraints/dataset.py

```diff
@@ -61,15 +77,32 @@ def __call__(self, value) -> DatasetParameter:
                 elif not self._installed and is_installed:
                     raise ValueError(f'{value} already exists locally')
             return DatasetParameter(value, value)
-        elif not isinstance(value, (str, PurePath)):
+        # anticipate what require_dataset() could handle and fail if we got
+        # something else
+        elif not isinstance(value, (str, PurePath, type(None))):
             raise TypeError(f"Cannot create Dataset from {type(value)}")
 
         from datalad.distribution.dataset import require_dataset
-        ds = require_dataset(
-            value,
-            check_installed=self._installed is True,
-            purpose=self._purpose,
-        )
+        try:
+            ds = require_dataset(
+                value,
+                check_installed=self._installed is True,
+                purpose=self._purpose,
+            )
+        except NoDatasetFound:
+            # mitigation of non-uniform require_dataset() behavior.
+            # with value == None it does not honor check_installed
+            # https://github.com/datalad/datalad/issues/7281
+            if self._installed is True:
+                # if we are instructed to ensure an installed dataset
+                raise
```

Comment:
`raise` or `from ... raise ValidationError` (if there is a ValidationError, if not then disregard the comment)?

Reply:
There is no ValidationError (yet?) -- only ValueError. I have no opinion ATM on what would be good to have here.

Reply:
This connects to open issue https://github.com/datalad/datalad-next/issues/198

This `ValidationError` should exist and it should be the workhorse for structured error reporting.